gallery_dl/extractor/facebook.py

```diff
@@ -0,0 +1,454 @@
+# -*- coding: utf-8 -*-
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extractors for https://www.facebook.com/"""
+
+from .common import Extractor, Message
+from .. import text, exception
+
+BASE_PATTERN = r"(?:https?://)?.*?facebook\.com"
+
+
+class FacebookExtractor(Extractor):
+    """Base class for Facebook extractors"""
+    category = "facebook"
+    root = "https://www.facebook.com"
+    directory_fmt = ("{category}", "{username}", "{title} ({set_id})")
+    filename_fmt = "{id}.{extension}"
+    archive_fmt = "{id}.{extension}"
+
+    set_url_fmt = root + "/media/set/?set={set_id}"
+    photo_url_fmt = root + "/photo/?fbid={photo_id}&set={set_id}"
+
+    def __init__(self, match):
+        Extractor.__init__(self, match)
+        self.match = match
+
+    def _init(self):
+        self.session.headers["Accept"] = "text/html"
+        self.session.headers["Sec-Fetch-Mode"] = "navigate"
+
+        self.fallback_retries = self.config("fallback-retries", 2)
+        self.sleep_429 = self.config("sleep-429", 5)
+        self.videos = self.config("videos", True)
+
+        self.author_followups = self.config("author-followups", False)
+
+    @staticmethod
+    def decode_all(txt):
+        return text.unescape(
+            txt.encode("utf-8").decode("unicode_escape")
+        ).replace("\\/", "/")
+
+    @staticmethod
+    def item_filename_handle(item):
+        item["filename"] = item.get("filename") or ""
+        if "." in item["filename"]:
+            item["name"], item["extension"] = item["filename"].rsplit(".", 1)
+        else:
+            item["name"] = item["filename"]
+            item["extension"] = ""
```

Comment:
Filenames are usually handled by [`text.nameext_from_url()`](https://github.com/mikf/gallery-dl/blob/d1ad97ae0c1417bbcabb84fba13ecd514ba11223/gallery_dl/text.py#L88)

Also, I prefer using `str.partiton()` over `str.split()` when splitting in two, but this is not that important.

Reply:
Should I just ditch the "name" keyword then?

Reply:
Yes. A file called `name.ext` should have `name` as `filename` and `ext` as `extension`. At least that's the usual convention.